### ISSUE TITLE
Astropy dev test failure for HGS->HCRS. 

### ISSUE THREAD

**Issue by nabobalis** [sunpy/sunpy#3066]
Currently with the latest version of astropy master. 

```python
def test_hgs_hcrs():
        # This test checks the HGS->HCRS transformation by transforming from HGS to
        # HeliocentricTrueEcliptic (HTE).  It will fail if there are errors in Astropy's
        # HCRS->ICRS or ICRS->HTE transformations.
    
        # Use published HGS coordinates in the Astronomical Almanac (2013), pages C6-C7
        obstime = Time('2013-01-28')
        earth_hgs = SkyCoord(0*u.deg, -5.73*u.deg, 0.9848139*u.AU, frame=HeliographicStonyhurst,
                             obstime=obstime)
    
        # Transform to HTE at observation-time equinox
        earth_hte = earth_hgs.transform_to(HeliocentricTrueEcliptic(equinox=obstime))
    
        # Validate against published values from the Astronomical Almanac (2013), page C6 per page E2
        # The dominant source of inaccuracy is the limited precision of the published B0 used above
>       assert quantity_allclose(earth_hte.lon, Angle('308d13m30.51s') - 180*u.deg, atol=5*u.arcsec)
E       assert False
E        +  where False = quantity_allclose(<Longitude 128.23007468 deg>, (<Angle 308.22514167 deg> - (180 * Unit("deg"))), atol=(5 * Unit("arcsec")))
E        +    where <Longitude 128.23007468 deg> = <SkyCoord (HeliocentricTrueEcliptic: equinox=2013-01-28 00:00:00.000, obstime=2013-01-28 00:00:00.000): (lon, lat, distance) in (deg, deg, AU)\n    (128.23007468, 0.00023427, 0.9848139)>.lon
E        +    and   <Angle 308.22514167 deg> = Angle('308d13m30.51s')
E        +    and   Unit("deg") = u.deg
E        +    and   Unit("arcsec") = u.arcsec
```

Need to figure out what's changed and if it's ok.

**Comment by Cadair** [comment]
ping @ayshih I will have a go at bisecting this later.

**Comment by Cadair** [comment]
Is this also failing on 0.9?

**Comment by ayshih** [comment]
<whimper>

**Comment by nabobalis** [comment]
Not a clue if it happens on 0.9.

**Comment by Cadair** [comment]
Ok so bisect reveals that the offending commit is https://github.com/astropy/astropy/commit/3d38c0e6b4866f5083d1945b9566a670305ec82b from https://github.com/astropy/astropy/pull/8394.

Looks like what we need to do is conditionally import the frame based on the version of astropy. I will ask @Juanlu001 for confirmation though.

**Comment by Cadair** [comment]
To be honest I am a little confused as to where the ecliptic transforms come into this at all.

**Comment by Juanlu001** [comment]
> To be honest I am a little confused as to where the ecliptic transforms come into this at all.

**Comment by Cadair** [comment]
oh :woman_facepalming: The test is what's broken and not the actual transform lol!

**Comment by Cadair** [comment]
@Juanlu001 The test seems to pass again if I swap to `MeanEcliptic` when was that frame added to astropy?

**Comment by Juanlu001** [comment]
@Cadair you said it yourself :) https://github.com/astropy/astropy/pull/8394

TL;DR: `TrueEcliptic` was, in fact, `MeanEcliptic`, so we corrected `True` frames and added `Mean` frames.

**Comment by Cadair** [comment]
Right so to support both 3.1 and 3.2 we need to conditionally import True for 3.1 and Mean for 3.2?

**Comment by Juanlu001** [comment]
Unfortunately yes :(

**Comment by Juanlu001** [comment]
```python
try:
    from astropy.coordinates import HeliocentricMeanEcliptic as HeliocentricEliptic
except ImportError:
    from astropy.coordinates import HeliocentricTrueEcliptic as HeliocentricEliptic
```

**Comment by ayshih** [comment]
Ha, sweet!  I confirm that the Astronomical Almanac provides Heliocentric Mean Ecliptic coordinates, so the test is in fact supposed to succeed with `Mean` and fail with `True`.  (I honestly was a little surprised back in the day that `HeliocentricTrueEcliptic` worked, and blissfully chose to look the other way at the time!)
